src/services/NavigationService.js

```diff
@@ -0,0 +1,36 @@
+import {NavigationActions, StackActions} from 'react-navigation';
+import {func} from 'prop-types';
+
+let _navigator;
+
+function setTopLevelNavigator(navigatorRef) {
+  _navigator = navigatorRef;
+}
+
+function navigate(routeName, params) {
+  _navigator.dispatch(
+    NavigationActions.navigate({
+      routeName,
+      params,
+    }),
+  );
+}
+
+function reset(name) {
```

Comment:
I think now it's okey, but in the future you will be had a lot of troubles with reset, because you reset navigation to 1 element into stack and other screens into stack will disappear.